mogp_emulator/GaussianProcessGPU.py

```diff
@@ -133,6 +140,60 @@ def interpret_nugget(nugget):
     # return info needed to set the nugget on the C++ object
     return nugget_type, nugget_size
 
+def get_prior_params(newpriors, inputs, n_corr, nugget_type):
+    """
+    Extract the parameters needed for the C++ DenseGP_GPU instance to 
+    create its GPPriors object.
+    Can accept either an existing GPPriors object, or the parameters needed
+    to construct one, or otherwise will create a default object.
+    :param newpriors: set of priors to use
+    :type newpriors: GPPriors or dict
+    :param inputs: the inputs of the GP
+    :type inputs: numpy array
+    :param n_corr: the number of correlation length parameters
+    :type n_corr: int
+    :param nugget_type: whether nugget is adaptive, fit, or fixed
+    "type nugget_type: LibGPGPU.nugget_type
+
+    :returns: dict of values needed to call the C++ constructor
+    :rtype: dict
+    """
+    if newpriors is None:
+        priors = GPPriors.default_priors(inputs, n_corr, nugget_type)
+    elif isinstance(newpriors, GPPriors):
+        priors = newpriors
+    else:
+        try:
+            priors = GPPriors(**newpriors)
+        except TypeError:
+            raise TypeError("Provided arguments for priors are not valid inputs " +
+                            "for a GPPriors object.")
+    # now return parameters needed to create corresponding C++ objects
+    prior_dict = {}
+    prior_dict["n_corr"] = n_corr
+    for prefix, prior in [("corr",priors.corr[0]),("cov",priors.cov),("nug",priors.nugget)]:        
+        if isinstance(prior, InvGammaPrior):
```

Comment:
Yep, I like this...  I ended up putting this logic with the type_dict and `prior_as_str` into a function `get_prior_params` that does this for a single prior (and renamed what was previously `get_prior_params` into `create_prior_params`).  
And rather than creating a dictionary, from which `_set_priors` picks out some values, this function now directly creates the arguments to be passed to the C++ interface (including, as mentioned above, more use of lists/vectors so that we can have different types of correlation length priors, and different numbers of parameters for each prior dist).